File: client/modules/native-list-interactions/ios/NativeListInteractionsModule.swift
import ExpoModulesCore

public final class NativeListInteractionsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeListInteractions")

    View(NativeListInteractionsView.self) {
      Prop("sectionsJson") { (view: NativeListInteractionsView, sectionsJson: String?) in
        view.updateSectionsJson(sectionsJson ?? "[]")
      }

      Prop("iosCategoryGestureMode") { (view: NativeListInteractionsView, mode: String?) in
        view.updateIOSCategoryGestureMode(mode)
      }

      Events(
        "onItemPress",
        "onMenuAction",
        "onDelete",
        "onReorder",
        "onToggleSwitch"
      )
    }
  }
}

File: client/modules/native-list-interactions/android/src/main/java/expo/modules/nativelistinteractions/NativeListInteractionsModule.kt
package expo.modules.nativelistinteractions

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeListInteractionsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeListInteractions")

    View(NativeListInteractionsView::class) {
      Prop("sectionsJson") { view: NativeListInteractionsView, sectionsJson: String? ->
        view.updateSectionsJson(sectionsJson ?: "[]")
      }

      Prop("iosCategoryGestureMode") { _: NativeListInteractionsView, _: String? ->
        // iOS-only compatibility prop.
      }

      Prop("contentInsetBottom") { view: NativeListInteractionsView, contentInsetBottom: Double? ->
        view.updateContentInsetBottom(contentInsetBottom ?: 0.0)
      }

      Events(
        "onItemPress",
        "onMenuAction",
        "onDelete",
        "onReorder",
        "onToggleSwitch",
        "onSectionExpandRequest"
      )
    }
  }
}
